security/fma-stage-1.md

```diff
@@ -0,0 +1,127 @@
+
+## Failure Modes and Recovery Paths
+
+### FM1: System Cannot Be Paused
+
+- **Description:** If the system's pause mechanism fails or is not properly implemented, the system cannot be paused in case of emergencies, potentially leading to continued operation during critical issues.
+- **Risk Assessment:** High impact, low likelihood
+- **Mitigations:** 
+  1. Testing of pause functionality
+  2. Clear documentation of pause procedures
+- **Detection:** 
+  - Monitoring of pause-related events
+- **Recovery Path(s)**: 
+  - If pause mechanism fails, immediate governance action may be required
+  - May require emergency upgrade to fix pause mechanism
+  - Consider implementing backup pause mechanisms
+
+### FM2: System Cannot Be Unpaused
+
+- **Description:** If the system's unpause mechanism fails, the system remains paused indefinitely, causing service disruption and potential loss of user access.
+- **Risk Assessment:** High impact, low likelihood
+- **Mitigations:**
+  1. Testing of unpause functionality
+  2. Clear documentation of unpause procedures
+- **Detection:**
+  - Monitoring of unpause-related events
+- **Recovery Path(s)**:
+  - If unpause mechanism fails, immediate governance action may be required
+  - May require emergency upgrade to fix unpause mechanism
+  - Consider implementing backup unpause mechanisms
+
+### FM3: Upgrade Process Bug
+
+- **Description:** If there is a bug in the upgrade process or implementation, it could lead to incorrect behavior, or complete system failure.
+- **Risk Assessment:** High impact, medium likelihood
+- **Mitigations:**
+  1. Comprehensive testing of upgrade process
+  2. Multiple deployments before mainnet
+  3. Refer to [fma-generic-contracts](https://github.com/ethereum-optimism/design-docs/blob/main/security/fma-generic-contracts.md), items applies to this case. Any new implementation must go through the audit and testing. Upgrades should not be regular in this contract and it should maintain minimal code. Upgrade procedures and keys must follows the proper security practices.
+- **Detection:**
+  - Post-upgrade monitoring and verification
+  - Verify proxy and implementation contracts through automated checks in the superchain-ops task
+- **Recovery Path(s)**:
+  - If upgrade fails, system may need to be upgraded again
+
+
+### FM4: Invalid identifier on pause
+- **Description:** If somehow the identifier used in the pause mechanism of the superchain config receives an invalid value, the pause mechanism will not be triggered for the right cluster of the superchain.
+- **Risk Assessment:** Medium impact, low likelihood
+- **Mitigations:**
+  1. Always call the pause function through the SystemConfig contract, that gets the identifier (EthLockbox) from the OptimismPortal2 contract.
+- **Detection:**
+  - Post-upgrade monitoring and verification
+- **Recovery Path(s)**:
+  1. Unpausing the invalid identifier
+  2. Pausing the correct identifier
+
```

Comment:
Other failure modes:
- Runbooks are outdated, people don't know how or when to use the new pause
- People change their lockbox or upgrade while paused, causing the pause to unpause itself